sky/utils/controller_utils.py

```diff
@@ -496,13 +511,17 @@ def get_controller_resources(
             #     - cloud: runpod
             #       accelerators: A40
             # In this case, we allow the controller to be launched on any cloud.
-            requested_clouds.clear()
+            requested_clouds_with_region_zone.clear()
             break
-    if not requested_clouds:
+    if not requested_clouds_with_region_zone:
         return {controller_resources_to_use}
     return {
-        controller_resources_to_use.copy(cloud=controller_cloud)
-        for controller_cloud in requested_clouds
+        controller_resources_to_use.copy(
+            cloud=clouds.CLOUD_REGISTRY.from_str(cloud_name),
+            region=region,
+            zone=zone)
+        for cloud_name, regions in requested_clouds_with_region_zone.items()
+        for region, zones in regions.items() for zone in zones
```

Comment:
As mentioned here, lets do explicit for loop instead?

https://github.com/skypilot-org/skypilot/pull/4053#discussion_r1797483419

Reply:
Fixed. Please review.